nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs

```diff
@@ -2301,35 +2304,63 @@ private async void ResponseBattleArenaAsync(ActionEvaluation<BattleArena> eval)
         }
 
         private async Task<(ArenaPlayerDigest myDigest, ArenaPlayerDigest enemyDigest)> GetArenaPlayerDigestAsync(
+            IAccountStateDelta states,
             Address myAvatarAddress,
             Address enemyAvatarAddress)
         {
-            var avatarStates = await Game.Game.instance.Agent.GetAvatarStates(new[] { myAvatarAddress, enemyAvatarAddress });
+            var avatarStates = await Game.Game.instance.Agent
+                .GetAvatarStates(new[] { myAvatarAddress, enemyAvatarAddress });
             var myAvatarState = avatarStates[myAvatarAddress];
             var enemyAvatarState = avatarStates[enemyAvatarAddress];
 
-            var (myItemSlotStates, myRuneSlotStates) = await myAvatarState.GetSlotStatesAsync();
-            var myItemSlotState =
-                myItemSlotStates.FirstOrDefault(x => x.BattleType == BattleType.Arena);
-            var myRuneSlotState =
-                myRuneSlotStates.FirstOrDefault(x => x.BattleType == BattleType.Arena);
-            var myRuneStates = (await myAvatarState.GetRuneStatesAsync())
-                .Where(x => myRuneSlotState.GetRuneSlot().Any(slot => x.RuneId == slot.RuneId))
-                .ToList();
+            var myItemSlotStateAddress = ItemSlotState.DeriveAddress(myAvatarAddress, BattleType.Arena);
+            var myItemSlotState = states.TryGetState(myItemSlotStateAddress, out List rawItemSlotState)
+                ? new ItemSlotState(rawItemSlotState)
+                : new ItemSlotState(BattleType.Arena);
+
+            var myRuneSlotStateAddress = RuneSlotState.DeriveAddress(myAvatarAddress, BattleType.Arena);
+            var myRuneSlotState = states.TryGetState(myRuneSlotStateAddress, out List rawRuneSlotState)
+                ? new RuneSlotState(rawRuneSlotState)
+                : new RuneSlotState(BattleType.Arena);
+
+            var myRuneStates = new List<RuneState>();
+            var myRuneSlotInfos = myRuneSlotState.GetEquippedRuneSlotInfos();
+            foreach (var address in myRuneSlotInfos.Select(info =>
+                RuneState.DeriveAddress(myAvatarAddress, info.RuneId)))
+            {
+                if (states.TryGetState(address, out List rawRuneState))
+                {
+                    myRuneStates.Add(new RuneState(rawRuneState));
+                }
+            }
```

Comment:
```suggestion
            }
            
            myAvatarState.UpdateEquipment(myItemSlotState.Equipments);
            myAvatarState.UpdateCostumes(myItemSlotState.Costumes);
```

In [BattleArena](https://github.com/planetarium/lib9c/blob/main/Lib9c/Action/BattleArena.cs#L341-L342), the avatarState will be updated equipments and costumes. Is this not necessary?
My and enemy avatar both.